### PR TITLE
PSP-6865 Property Management summary

### PR DESCRIPTION
**Pull request by asanchezr**
Screenshot 2023-09-28 160625

### CODE REVIEW

File: source/backend/api/Areas/Property/Controllers/PropertyManagementController.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            _propertyService = propertyService;
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Get the property management information for the property with 'propertyId'.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("{propertyId}/management")]
+        [HasPermission(Permissions.PropertyView)]
```
Comment (devinleighsmith) [inline source/backend/api/Areas/Property/Controllers/PropertyManagementController.cs:45]: This should probably also require the new view-management claim should it not?

File: source/backend/api/Areas/Property/Controllers/PropertyManagementController.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            return new JsonResult(_propertyService.GetPropertyManagement(propertyId));
+        }
+
+        /// <summary>
+        /// Update the specified property management information.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPut("{propertyId}/management")]
+        [HasPermission(Permissions.PropertyEdit)]
```
Comment (devinleighsmith) [inline source/backend/api/Areas/Property/Controllers/PropertyManagementController.cs:60]: similar comment here.

File: source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/detail/summary/PropertyManagementDetailContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    getPropertyManagement: { execute: getPropertyManagement, loading },
+  } = usePropertyManagementRepository();
+
+  const fetchPropertyManagement = useCallback(async () => {
+    if (!property.id) {
+      return;
+    }
+    const response = await getPropertyManagement(property.id);
+    if (response) {
+      setPropertyManagement(response);
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/detail/summary/PropertyManagementDetailContainer.tsx:40]: strictly speaking I think you should either be checking if the component is still mounted here, or skip using this state and use the data object that is returned from the repository.

File: source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/update/summary/__snapshots__/PropertyManagementUpdateContainer.test.tsx.snap [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`PropertyManagementUpdateContainer component renders as expected 1`] = `
+<DocumentFragment>
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/update/summary/__snapshots__/PropertyManagementUpdateContainer.test.tsx.snap:4]: Might need to wait for the page to load here.

File: source/frontend/src/features/mapSideBar/acquisition/tabs/checklist/detail/__snapshots__/AcquisitionChecklistView.test.tsx.snap [hunk 5]
```diff
@@ -1754,6 +1754,51 @@ exports[`AcquisitionChecklistView component renders as expected 1`] = `
         </div>
       </div>
     </div>
+    <div
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/acquisition/tabs/checklist/detail/__snapshots__/AcquisitionChecklistView.test.tsx.snap:1757]: why is the acquisition checklist being modified by this PR?

**Comment by devinleighsmith** [comment]
THe property purpose appears to default to other but I don't see that in the requirements.

**Comment by devinleighsmith** [comment]
Not sure, but some of the implemented functionality appears to correlate to a crossed out section in confluence.

image

not sure if that logic should be in there or not.

**Review by devinleighsmith — APPROVED** [review]

File: source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/update/summary/__snapshots__/PropertyManagementUpdateContainer.test.tsx.snap (hunk 4 above)
Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/update/summary/__snapshots__/PropertyManagementUpdateContainer.test.tsx.snap:4]: Will actually remove this snapshot as it is for the container test that renders an empty view which is why it shows like that. The view does have an snapshot test

**Comment by asanchezr** [comment]
> Not sure, but some of the implemented functionality appears to correlate to a crossed out section in confluence.
> 
> image
> 
> not sure if that logic should be in there or not.

hmm - this was done after I implemented and demoed the story to the business. So I assume any changes will come in an additional story. Would be ideal if BAs used confluence versioning properly

File: source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/detail/summary/PropertyManagementDetailContainer.tsx (hunk 3 above)
Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/detail/summary/PropertyManagementDetailContainer.tsx:40]: Will change - I was using the same pattern as Manuel on this property contact PR

**Comment by asanchezr** [comment]
> THe property purpose appears to default to other but I don't see that in the requirements.

Sorry, I don't follow. AFAIK, the property purposes (plural) is an array (ie multi-select) and it default to the empty array. What is implemented (and called out in Confluence) is that if that array  contains the code OTHER, then the **additional details** textarea is required

File: source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/detail/summary/PropertyManagementDetailContainer.tsx (hunk 3 above)
Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/detail/summary/PropertyManagementDetailContainer.tsx:40]: I will change this in my next PR

**Pull request merged**